test/elixir_analyzer/test_suite/file_sniffer_test.exs

```diff
@@ -147,40 +153,90 @@ defmodule ElixirAnalyzer.ExerciseTest.FileSnifferTest do
               "image/gif"
           end
         end
+      end,
+      defmodule FileSniffer do
+        @exe_type "application/octet-stream"
+        @bmp_type "image/bmp"
+        @png_type "image/png"
+        @jpg_type "image/jpg"
+        @gif_type "image/gif"
+
+        def type_from_binary(<<"\dELF", _rest::binary>>), do: @exe_type
+        def type_from_binary(<<"BM", _rest::binary>>), do: @bmp_type
+        def type_from_binary(<<"\x89PNG\r\n\x1A\n", _rest::binary>>), do: @png_type
+        def type_from_binary(<<"\xFF\xD8\xFF", _rest::binary>>), do: @jpg_type
+        def type_from_binary(<<"GIF", _rest::binary>>), do: @gif_type
+      end,
+      defmodule FileSniffer do
+        def type_from_binary(<<signature::binary-size(2), _rest::binary>>)
+            when <<0x42, 0x4D>> == signature,
+            do: "image/bmp"
+
+        def type_from_binary(<<signature::binary-size(8), _rest::binary>>)
+            when <<0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A>> == signature,
+            do: "image/png"
+
+        def type_from_binary(<<signature::binary-size(3), _rest::binary>>)
+            when <<0xFF, 0xD8, 0xFF>> == signature,
+            do: "image/jpg"
+
+        def type_from_binary(<<signature::binary-size(3), _rest::binary>>)
+            when <<0x47, 0x49, 0x46>> == signature,
+            do: "image/gif"
+
+        def type_from_binary(<<signature::binary-size(4), _rest::binary>>)
+            when <<0x7F, 0x45, 0x4C, 0x46>> == signature,
+            do: "application/octet-stream"
+
+        def type_from_binary(_binary), do: nil
+      end,
+      defmodule FileSniffer do
+        def type_from_binary(<<66, 77, 30, _::binary>>), do: type_from_extension("bmp")
+
+        def type_from_binary(<<71, 73, 70, 56, 57, 97, _::binary>>),
+          do: type_from_extension("gif")
+
+        def type_from_binary(<<255, 216, 255, 219, _::binary>>), do: type_from_extension("jpg")
+
+        def type_from_binary(<<137, 80, 78, 71, 13, 10, 26, 10, _::binary>>),
+          do: type_from_extension("png")
+
+        def type_from_binary(<<127, 69, 76, 70, _::binary>>), do: type_from_extension("exe")
+        def type_from_binary(file), do: type_from_extension(file)
       end
     ]
   end
 
-  test_exercise_analysis "doesn't use pattern matching",
-    comments: [Constants.file_sniffer_use_pattern_matching()] do
+  test_exercise_analysis "doesn't use <<>> nor ::",
```

Comment:
The comment should be triggered if the solution only uses one of the forms right? Should we add tests to check that?

Reply:
I added a test with just `<<>>` but I don't think I can add a test with just `::`. It's used for defining binary sizes, so you can't really use it without using the binary special form...